trio/tests/test_socket.py

```diff
@@ -389,55 +393,82 @@ def gai_without_v4mapped_is_buggy():  # pragma: no cover
-        got = await sock4._resolve_local_address((null, 80))
-        assert got == ("0.0.0.0", 80)
-        got = await sock4._resolve_remote_address((null, 80))
-        assert got == ("127.0.0.1", 80)
-
-        sock6 = tsocket.socket(family=tsocket.AF_INET6)
-        got = await sock6._resolve_local_address((null, 80))
-        assert got == ("::", 80, 0, 0)
-
-        got = await sock6._resolve_remote_address((null, 80))
-        assert got == ("::1", 80, 0, 0)
+        sock = tsocket.socket(family=socket_type)
+        got = await sock._resolve_local_address((null, 80))
+        assert got == (addrs.bind_all, 80, *addrs.extra)
+        got = await sock._resolve_remote_address((null, 80))
+        assert got == (addrs.localhost, 80, *addrs.extra)
 
     # AI_PASSIVE only affects the wildcard address, so for everything else
     # _resolve_local_address and _resolve_remote_address should work the same:
-    for res in ["_resolve_local_address", "_resolve_remote_address"]:
-
-        async def s4res(*args):
-            return await getattr(sock4, res)(*args)
-
-        async def s6res(*args):
-            return await getattr(sock6, res)(*args)
+    for resolver in ["_resolve_local_address", "_resolve_remote_address"]:
 
-        assert await s4res(("1.2.3.4", "http")) == ("1.2.3.4", 80)
-        assert await s6res(("1::2", "http")) == ("1::2", 80, 0, 0)
+        async def res(*args):
+            return await getattr(sock, resolver)(*args)
 
-        assert await s6res(("1::2", 80, 1)) == ("1::2", 80, 1, 0)
-        assert await s6res(("1::2", 80, 1, 2)) == ("1::2", 80, 1, 2)
+        assert await res((addrs.arbitrary,
+                          "http")) == (addrs.arbitrary, 80, *addrs.extra)
+        if v6:
+            assert await res(("1::2", 80, 1)) == ("1::2", 80, 1, 0)
+            assert await res(("1::2", 80, 1, 2)) == ("1::2", 80, 1, 2)
 
-        # V4 mapped addresses resolved if V6ONLY is False
-        sock6.setsockopt(tsocket.IPPROTO_IPV6, tsocket.IPV6_V6ONLY, False)
-        assert await s6res(("1.2.3.4", "http")) == ("::ffff:1.2.3.4", 80, 0, 0)
+            # V4 mapped addresses resolved if V6ONLY is False
+            sock.setsockopt(tsocket.IPPROTO_IPV6, tsocket.IPV6_V6ONLY, False)
+            assert await res(("1.2.3.4",
+                              "http")) == ("::ffff:1.2.3.4", 80, 0, 0)
 
         # Check the <broadcast> special case, because why not
-        await s4res(("<broadcast>", 123)) == ("255.255.255.255", 123)
-        await s6res(("<broadcast>", 123)) == ("::ffff:255.255.255.255", 123)
+        assert await res(("<broadcast>",
+                          123)) == (addrs.broadcast, 123, *addrs.extra)
 
         # But not if it's true (at least on systems where getaddrinfo works
         # correctly)
-        if not gai_without_v4mapped_is_buggy():  # pragma: no branch
-            sock6.setsockopt(tsocket.IPPROTO_IPV6, tsocket.IPV6_V6ONLY, True)
+        if v6 and not gai_without_v4mapped_is_buggy():
```

Comment:
Azure allegedly [supports ipv6](https://azure.microsoft.com/en-us/updates/ipv6-for-azure-vms/), it might be worth checking if the pipelines linux vms have it.

Reply:
Yep, apparently our CI can create an IPv6 socket but not bind it, while the system I'm using that prompted this PR can't even create one. I'm splitting out the pytest marks to be aware of that distinction, which should resolve this.

Reply:
Ah, interesting! Do you happen to know what the configuration difference between those machines is? It be nice to know if there's some way to recreate the different configurations in CI...

Reply:
The machine that prompted this PR has `ipv6.disable=1` in the Linux kernel command line. I'm not sure if there's a less invasive way to produce the same result or not, but I suspect it's not the sort of thing a CI service would let you configure.

Reply:
Ah, yeah, it looks like `ipv6.disable` on the command line is documented here: https://www.kernel.org/doc/Documentation/networking/ipv6.txt

> No IPv6 addresses will be added to interfaces, and it will not be possible to open an IPv6 socket.
> A reboot is required to enable IPv6.

OTOH there's the `disable_ipv6` sysctl, documented here: https://www.kernel.org/doc/Documentation/networking/ip-sysctl.txt

But that works on an interface-by-interface basis, so I guess it makes sense that if you use the sysctl to disable it on all interfaces, then you can still create the sockets but then they fail to `bind`. Oh well, good to know I guess.